src/omnicore/rpc.cpp

```diff
@@ -226,6 +226,56 @@ UniValue omni_getuniquetokenowner(const UniValue& params, bool fHelp)
     return rpcObj;
 }
 
+// displays all the ranges and their addresses for a property
+UniValue omni_getuniquetokenranges(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "omni_getuniquetokenranges propertyid\n"
+            "\nReturns the ranges and their addresses for a unique token property.\n"
+            "\nArguments:\n"
+            "1. propertyid           (number, required) the property identifier\n"
+            "\nResult:\n"
+            "[                           (array of JSON objects)\n"
+            "  {\n"
+            "  \"address\" : \"address\", (string) the address\n"
+            "  \"uniquetokenstart\" : n,  (number) the first token in this range\n"
+            "  \"uniquetokenend\" : n,    (number) the last token in this range\n"
+            "  \"amount\" : n             (number) the amount of tokens in the range\n"
+            "  },\n"
+            "  ...\n"
+            "]\n"
+            "\nExamples:\n"
+            + HelpExampleCli("omni_getuniquetokenranges", "1")
+            + HelpExampleRpc("omni_getuniquetokenranges", "1")
+        );
+
+    uint32_t propertyId = ParsePropertyId(params[0]);
+
+    RequireExistingProperty(propertyId);
+
+    UniValue response(UniValue::VARR);
+
+    std::vector<std::pair<std::string,std::pair<int64_t,int64_t> > > rangeMap = p_utdb->GetUniqueTokenRanges(propertyId);
+
+    for (std::vector<std::pair<std::string,std::pair<int64_t,int64_t> > >::iterator it = rangeMap.begin(); it!= rangeMap.end(); ++it) {
+        std::pair<std::string,std::pair<int64_t,int64_t> > entry = *it;
+        std::string address = entry.first;
+        std::pair<int64_t,int64_t> range = entry.second;
+        int64_t amount = (range.second - range.first) + 1;
```

Comment:
This is done a few times now, maybe we could add a function to calculate the amount? Even if it may seem a bit lame to do so, something like the `+ 1` is probably error prone.

Reply:
You're right - if a single address owned max INT64 tokens `amount` would overflow.  Do you think it's safe to check if `range.first || range.second` are less than max INT64?

Reply:
If done is independent checks, it should be fine. :)